### PR TITLE
WIP: Use a monorepo to host styleguidist

### PR DESCRIPTION
**Pull request by elevatebart**
Objective: enable splitting react-styleguidist into multiple packages without losing any developer experience.

- [x] Move all code into packages/react-styleguidist
- [x] Use yarn workspaces to manage dependencies
- [x] Make compile + tests + cypress green
- [x] Review ci to use lerna
- [x] Make sure all scripts are still working the same

NOTA: I voluntarily left out all the changes regarding extracting the core of styleguidist from the rest. I fell they would confuse proofreading.

### CODE REVIEW

**Comment by sapegin** [comment]
Thanks! Could you add a list of all the changes (and possible consequences ;-) to the PR description? I'll also mark it as WIP for now.

File: package.json [hunk 1]
```diff
@@ -123,43 +37,43 @@
...
     "style-loader": "^0.23.1",
     "url-loader": "^1.1.2",
-    "webpack": "^4.26.0",
     "webpack-cli": "^3.1.2"
   },
   "scripts": {
-    "pretest": "npm run lint",
-    "test": "npm run test:jest",
-    "posttest": "npm run format",
+    "pretest": "yarn lint",
```
Comment (okonet) [inline package.json]: No need for that change afaik. This can be problematic on the CI

Comment (elevatebart) [reply package.json]: absolutely right, I removed it

**Comment by okonet** [comment]
LGTM!

**Comment by elevatebart** [comment]
Hey @sagepin, 
I noticed that the next branch is out of date with master. 
Should I rebase this branch on master and make a new PR?

**Comment by sapegin** [comment]
Yeah, sounds good. We don't really have `next` branch all the time, just for major releases.

**Pull request closed without merging**